src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -73,6 +73,19 @@ using namespace GafferScene;
 namespace
 {
 
+/// \todo Make this into an API for nodes to register the attributes and corresponding
+/// default values they will make available. This is needed to get the value of an
+/// attribute that _could_ exist if the user activates it, allowing it to be discovered
+/// in clients of history related APIs such as `AttributeInspector`.
+typedef std::unordered_map<std::string, const IECore::Data *> AttributeRegistry;
```

Comment:
I was thinking it might be a tiny bit faster to not increment the counter of a shared pointer, but making the ownership explicit does seem nice.

Fixed in ece26aaccb86c74f9cc58b10d3db590d01d70c34